### PR TITLE
travis update (osx + linux)

### PR DESCRIPTION
**Pull request by tooomm**
## Short roundup of the initial problem
- Caching all homebrew directory resulted in a very big cache file which took a long time to download, validate, check for changes and upload again (~50s longer than the one additional "brew update" run)

## What will change with this Pull Request?
- Remove homebrew caching
- Update to last xcode image with 10.11 support (Xcode 8 GM):
https://docs.travis-ci.com/user/reference/osx/#OS-X-Version
- Rename `.travis` folder for scripts to `.ci` to place appveyor stuff there too
- Limit `BUILDTYPE=Release` builds to jobs from `master` branch which are no PR's to prevent unneeded builds and return a useful result more quickly in the GitHub interface.
Travis only allows a certain number of jobs that run in parallel, this reduces our own queue we produce.
[Release = Debug just without tests](https://github.com/Cockatrice/Cockatrice/blob/tooomm-travis_osx/.ci/travis-compile.sh#L21) -> so we should not miss out on something
    - Docs: https://docs.travis-ci.com/user/conditional-builds-stages-jobs/#Conditional-Jobs
- Little tweaks

<br>

While this slightly helps, the big speed improvement for mac builds has to come with https://github.com/Cockatrice/Cockatrice/issues/2982

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,64 +1,48 @@
...
-        - qttools5-dev
-        - qttools5-dev-tools
-        - qtmultimedia5-dev
-        - libqt5multimedia5-plugins
-        - libqt5svg5-dev
-        - libqt5sql5-mysql
+    if: branch = master
   - os: osx
-    osx_image: xcode7.3
+    osx_image: xcode8
```
Comment (ctrlaltca) [inline .travis.yml]: Updating to xcode 8 means cockatrice will only run on osx => 10.12 sierra

Comment (tooomm) [reply .travis.yml]: As said above:
>Update to **last xcode image with 10.11 support** (Xcode 8 GM):
>https://docs.travis-ci.com/user/reference/osx/#OS-X-Version

from the [build log](https://travis-ci.org/Cockatrice/Cockatrice/jobs/331430456#L4):
`instance: 06841a6e-1629-4a9d-81d1-42a7e6d7b3d6 travis-ci-osx10.11-xcode8-1479248015`

Only starting with `xcode8.3` onwards it'll be osx =>10.12!

<br>

Is it worth to consider switchting to 10.12 yet? It cuts off some older hardware and 10.11 is just ~2 years old I think.

Comment (ctrlaltca) [reply .travis.yml]: Reading Apple's docs it says that xcode 8 only installs the SDK for macOS >=10.12, but probably travis handles this different. Sorry for the noise.

**Comment by tooomm** [comment]
Expanded the job conditions for `Release` builds to not run on PR's since every PR is build against master.
It'll still run on every merge to our master, or manually triggered builds on master like tagging or rerunning travis. Therefore deploying (which only happens from `Release` builds) is not affected.

Before:
before

After:
after

**Pull request merged**